backend/src/main.py

```
@@ -82,10 +95,14 @@ def internal_error(error):
-    # asyncio.run(mempool_monitor_service.start_monitoring()) # This needs to be run in a separate thread or managed by eventlet
-    # For eventlet, tasks are usually spawned directly
-    socketio.start_background_task(mempool_monitor_service.start_monitoring)
+    # Start background thread for asyncio
+    t = threading.Thread(target=run_async_loop, args=(bg_loop,), daemon=True)
+    t.start()
+
+    # Schedule background tasks in the async loop
+    asyncio.run_coroutine_threadsafe(mempool_monitor_service.start_monitoring(), bg_loop)
+
+    # Start Flask-SocketIO
     socketio.run(app, host='0.0.0.0', port=5000)
```

Ruff (0.14.14)

[error] Possible binding to all interfaces

(S104)